analyzers/src/SonarAnalyzer.CSharp/Rules/CommentsShouldNotBeEmpty.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Text;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class CommentsShouldNotBeEmpty : CommentsShouldNotBeEmptyBase<SyntaxKind>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
+
+    protected override bool IsValidTriviaType(SyntaxTrivia trivia)
+        => Language.SyntaxKind.CommentTrivia.Contains(trivia.Kind());
```

Comment:
Are you sure? `.Kind()` is an extension method, implemented twice in VB and C# 
`CodeAnalysis` DLLs.

The implementation looks like:
```csharp
        public static SyntaxKind Kind(this SyntaxTrivia trivia)
        {
            int rawKind = trivia.RawKind;
            if (!IsVisualBasicKind(rawKind))
            {
                return SyntaxKind.None;
            }

            return (SyntaxKind)rawKind;
        }
```
This^ is for VB, for C# it swaps `IsVisualBasicKind` with `IsCSharpKind`.

It looks a bit funky, because they have the same name, but the implementation is ostensibly different between C# and VB.

Reply:
SyntaxKind is an enum and as such just a number. You can do this 
```suggestion
    protected override bool IsValidTriviaType(SyntaxTrivia trivia)
        => Language.SyntaxKind.CommentTrivia.Contains((TSyntaxKind)trivia.RawKind);
```
in the base class.

Reply:
I cannot cast the `int` `RawKind` to a `TSyntaxKind`.
I tried adding some constraints, like `struct, System.Enum` but i still get the error.

Any ideas?